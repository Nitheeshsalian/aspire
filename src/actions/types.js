export default TYPES = {
    INCREMENT_LIMIT: "INCREMENT_LIMIT",
    RESET_LIMIT: "RESET_LIMIT"
}