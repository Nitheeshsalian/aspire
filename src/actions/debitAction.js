import TYPES from './types'

export function updateLimit(payload) {
    return {
        type: TYPES.INCREMENT_LIMIT,
        payload: payload
    }
}