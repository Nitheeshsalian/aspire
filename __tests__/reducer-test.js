/**
 * @format
 */

import 'react-native';
import React from 'react';
import configureStore from 'redux-mock-store';
import {updateLimit} from '../src/actions/debitAction';
const mockStore = configureStore();
const store = mockStore();

//Mock font awesome
jest.mock('@fortawesome/react-native-fontawesome', () => ({
  FontAwesomeIcon: '',
}));

describe('Reducer mock', () => {
  test('Dispatches the correct action and payload', () => {
    const expectedActions = [
      {
        payload: 5000,
        type: 'INCREMENT_LIMIT',
      },
    ];

    store.dispatch(updateLimit(5000));
    expect(store.getActions()).toEqual(expectedActions);
  });

  test('Match snapshot', () => {
    store.dispatch(updateLimit(5000));
    expect(store.getActions()).toMatchSnapshot();
  });
});
