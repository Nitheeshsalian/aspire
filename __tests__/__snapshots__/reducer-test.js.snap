// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Reducer mock Match snapshot 1`] = `
Array [
  Object {
    "payload": 5000,
    "type": "INCREMENT_LIMIT",
  },
  Object {
    "payload": 5000,
    "type": "INCREMENT_LIMIT",
  },
]
`;
